sketch/src/sketch/actions/plugin.js

```diff
@@ -69,7 +69,11 @@ export function perform(condition, layer, params) {
   setCommandParamsToMetadata(layer, params)
 
   // run the command
-  NSApp.delegate().runPluginCommand(command)
+  if (NSApp.delegate().runPluginCommand_fromMenu_context_) {
+    NSApp.delegate().runPluginCommand_fromMenu_context_(command, null, {})
```

Comment:
In theory we might use the `context` argument here to pass the same command params as we do via `userInfo`, but it will suffer from the same serialization limitations, so I wouldn't bother.